apps/rush-lib/src/data/VersionMismatchFinder.ts

```diff
@@ -97,6 +102,17 @@ export class VersionMismatchFinder {
     }
   }
 
+  private _isVersionAllowedAlternative(
+    dependency: string,
+    version: string): boolean {
+
+    const allowedAlternatives: ReadonlyArray<string> | undefined = this._allowedAlternativeVersion.get(dependency);
+    if (allowedAlternatives && allowedAlternatives.indexOf(version) > -1) {
```

Comment:
if you do either then you dont need this check

Reply:
Much nicer!